### PR TITLE
Add prose for typing rule of CAUGHTadm

### PR DESCRIPTION
**Pull request by ioannad**
Fixes related .. todo:: section.

Also slightly adjusted notation to match the one in the [formal overview document](https://github.com/WebAssembly/exception-handling/blob/main/proposals/exception-handling/Exceptions-formal-overview.md).

### CODE REVIEW

**Review by aheejin — APPROVED** [review]

File: document/core/appendix/properties.rst [hunk 1]
```diff
@@ -646,14 +646,25 @@ To that end, all previous typing judgements :math:`C \vdash \X{prop}` are genera
 :math:`\CAUGHTadm\{\tagaddr~\val^\ast\}~\instr^\ast~\END`
 .........................................................
 
-.. todo::
-   Add prose.
+* The :ref:`external tag value <syntax-externval>` :math:`\EVTAG~\tagaddr` must be :ref:`valid <valid-externval-tag>` with some :ref:`external tag type <syntax-externtype>` :math:`\ETTAG~[t_0^\ast] \to []`.
+
+* The :ref:`values <syntax-val>` :math:`\val^\ast` must be of type :math:`[t_0^\ast]`.
```
Comment (rossberg) [inline document/core/appendix/properties.rst:651]: I think this isn't defined as written, and requires a loop over the individual values.

File: document/core/appendix/properties.rst [hunk 2]
```diff
@@ -646,14 +646,25 @@ To that end, all previous typing judgements :math:`C \vdash \X{prop}` are genera
...
+
+* Let :math:`C'` be the same :ref:`context <context>` as :math:`C`, but with the label :math:`(\LCATCH~[t^\ast])` prepended to the |CLABELS| vector.
```
Comment (rossberg) [inline document/core/appendix/properties.rst:653]: This seems wrong. Note that the typing rule requires that there already is a labeltype t* at the front of the labels (which you need to check!), and this is just adjusting it.

File: document/core/appendix/properties.rst [hunk 3]
```diff
@@ -646,14 +646,25 @@ To that end, all previous typing judgements :math:`C \vdash \X{prop}` are genera
...
+
+* Under context :math:`C'`,
+  the instruction sequence :math:`\instr^\ast` must be :ref:`valid <valid-instr-seq>` with type :math:`[] \to [t^\ast]`.
+
+* Let :math:`C''` be the same :ref:`context <context>` as :math:`C`, but with the label :math:`[t^\ast]` prepended to the |CLABELS| vector.
+
+* Then the compound instruction is valid under context :math:`C''` with type :math:`[] \to [t^\ast]`.
```
Comment (rossberg) [inline document/core/appendix/properties.rst:660]: Ah, I see, what you are trying. But the context C is an input, you cannot construct another one here. If there are constraints on it, those have to be checked.

Comment (ioannad) [reply document/core/appendix/properties.rst:660]: @rossberg, looking for past review comments I had forgotten I found this, which was the reason for the (still misplaced) [change you found in PR#241](https://github.com/WebAssembly/exception-handling/pull/241#discussion_r1022935612). How should this prose and rule look like after all?

Comment (rossberg) [reply document/core/appendix/properties.rst:660]: The prose needs to first check the topmost entry of its input context C, and then construct C' from it by updating that, and check the body under that C'.

Comment (ioannad) [reply document/core/appendix/properties.rst:660]: @rossberg, I made PR #244 to match your description, but I'm not entirely sure about the notation I used.

**Pull request merged**